### PR TITLE
Add scheduler-service to support #Jira-SAV-2229

### PR DESCRIPTION
**Pull request by Bashar200**

### CODE REVIEW

File: src/routes/route.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const reqController = require('./routeController');
+const validate = require('../validate');
+
+router.post(
+  '/create',
+  validate(reqValidation.createSchedulerEvent),
+  reqController.createEvent,
+);
+router.get(
+  '/fetch',
```
Comment (GarvitBansal22) [inline src/routes/route.js:14]: suggestion: use /get-events instead of /fetch

File: src/routes/route.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+
+const router = express.Router();
+const reqValidation = require('../models/model');
+const reqController = require('./routeController');
+const validate = require('../validate');
+
+router.post(
+  '/create',
```
Comment (GarvitBansal22) [inline src/routes/route.js:9]: suggestion: use /event

File: src/routes/route.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  validate(reqValidation.createSchedulerEvent),
+  reqController.createEvent,
+);
+router.get(
+  '/fetch',
+  validate(reqValidation.getEvents),
+  reqController.fetchEvents,
+);
+router.put(
+  '/update',
```
Comment (GarvitBansal22) [inline src/routes/route.js:19]: suggestion: use /event instead of /update

File: src/routes/route.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  validate(reqValidation.updateEvent),
+  reqController.updateEvent,
+);
+router.delete(
+  '/delete',
```
Comment (GarvitBansal22) [inline src/routes/route.js:24]: suggestion: use /event instead of /delete

File: src/routes/route.js (hunk 2 above)
Comment (Bashar200) [reply src/routes/route.js:10]: can use different endpoint approach

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        "browser": false,
+        "commonjs": true,
+        "es2021": true
+    },
+    "extends": "airbnb-base",
+    "parserOptions": {
+        "ecmaVersion": "latest"
+    },
+    "rules": {
+        "no-console": 1,
```
Comment (mehul-mittal-credgenics) [inline .eslintrc.json:12]: it should be error level

**Review by mehul-mittal-credgenics — CHANGES_REQUESTED** [review]

File: src/jobs/jobDefinition.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    bodyCallBack,
+  } = payload;
+
+  try {
+    logger.info(`event triggered for ${eventName || job.attrs.data.eventName}`);
+    url = url || job.attrs.data.url;
+    bodyCallBack = bodyCallBack || job.attrs.data.bodyCallBack;
+    headerCallBack = headerCallBack || job.attrs.data.headerCallBack;
+    const response = await axios.post(url, bodyCallBack, { headers: headerCallBack });
+    logger.info(`Task API output:: ${JSON.stringify(response.data)} Event Name:: ${eventName}`);
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply src/jobs/jobDefinition.js:18]: converted to debug

File: src/settings.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import 'dotenv/config';
+
+const {
+  ENV, BASE_ROUTE, SERVICE_NAME, DB_NAME, MONGO_URI, LOG_LEVEL = 'INFO', TIMEOUT_MONGO = +TIMEOUT_MONGO, PORT = +PORT, MAX_CONCURRENCY = +MAX_CONCURRENCY, MAX_POOL_SIZE = 10,
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply src/settings.js:4]: done

File: .eslintrc.json (hunk 5 above)
Comment (Bashar200) [reply .eslintrc.json]: done

**Pull request closed without merging**